### PR TITLE
[Api/Single] invalid output if error occurs

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. allocate output buffer after invoke is done.
2. allow one buffer in appsink and drop old buffer.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single.c [hunk 1]
```diff
@@ -442,6 +437,16 @@ ml_single_invoke (ml_single_h single,
...
 
+  /* Allocate output buffer */
+  status = ml_tensors_data_create (&single_h->out_info, output);
+  if (status != ML_ERROR_NONE) {
+    ml_loge ("Failed to allocate the memory block.");
+    *output = NULL;
+    return status;
+  }
+
+  result = (ml_tensors_data_s *) (*output);
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single.c]: Why was this code block moved?

Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c]: When this function returns error (ML_ERROR_TIMED_OUT in line 437), the output-handle is allocated. So I moved these block after retrieving the buffer from appsink.

Comment (myungjoo) [reply api/capi/src/nnstreamer-capi-single.c]: Then you need
```
gst_sample_unref (sample);
```

Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c]: 1. sample unref
Right, I added gst_sample_unref() if failed to allocate mem block.

2. buffer
gst_app_src_push_buffer() takes ownership of the buffer.

Comment (myungjoo) [reply api/capi/src/nnstreamer-capi-single.c]: Ok. LTGM now.

Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c]: add buffer_unref if failed to push buffer to appsrc (if refcount is 0 already, it will do nothing.)

**Pull request merged**